### ISSUE TITLE
[Tracking] - Prysm v2.0.0 Release

### ISSUE THREAD

**Issue by rauljordan** [prysmaticlabs/prysm#9166]
## Background

The Altair hard fork is around the corner. This network upgrade marks an important milestone for Ethereum proof of stake and is also a unique opportunity for a mandatory, major version upgrade of the Prysm software. This means we have a special chance to do some breaking changes to Prysm that will improve the overall developer and user experience. This issue is meant to track the changes we want to include in v2 at a high-level without going into too much detail.

Most of these changes should be done until _after_ we have finished the Altair changes to the beacon chain to avoid creating conflicts and further issues.

## Critical Features
- [x] Merge in all Altair changes to make Prysm compatible
- [x] Complete and merge our optimized slasher implementation

## Deprecated Code Removal
- [x] Remove all deprecated protobufs
- [x] Delete slasher from the repo
- [x] Remove all deprecated flags and packages

## Documentation and Comms
- [x] Complete documentation versioning for v2, see [here](https://docusaurus.io/docs/versioning)
- [x] Announce changelog ahead of time to everyone running Prysm

## Package Restructuring 
- [x] Remove all extraneous packages and tools that are not directly relevant to running a beacon node / validator client into https://github.com/prysmaticlabs/periphery
- [x] Remove our `contracts/` package to reduce confusion as they are not the canonical eth2 contracts. If needed, they can be moved into the periphery
- [x] Deconstruct all the code in shared/ into top-level, organized packages by proper namespace and grouping according to  [this doc](https://hackmd.io/@prysmaticlabs/repo-restructure-v2)

### Extras
- [x] Use a single port for `/eth/v1`'s grpc-gateway and API Middleware endpoints
- [ ] Move eth2-types into Prysm

**Comment by rkapka** [comment]
What about our protoc plugin? It's not useful outside of Prysm, right?

**Comment by rkapka** [comment]
Also what about majority voting? Do we remove it all together in v2?

**Comment by prestonvanloon** [comment]
Can we update our reference implementation for remote signer to support altair changes prior to v2 release?

**Comment by rauljordan** [comment]
eth2 types is out of scope for this release. We are ready and already trying out a release candidate on testnets
